src/networkwatcher.cpp

```diff
@@ -170,5 +170,8 @@ void NetworkWatcher::notificationClicked(NotificationHandler::Message message) {
 }
 
 QNetworkInformation::Reachability NetworkWatcher::getReachability() {
-  return QNetworkInformation::instance()->reachability();
+  if (QNetworkInformation::instance()) {
+    return QNetworkInformation::instance()->reachability();
+  }
+  return QNetworkInformation::Reachability::Unknown;
```

Comment:
Why was a different NetworkWatcher added for each platform? (Instead of using https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8834?)

Reply:
It follows the same pattern as `getCurrentTransport()`, anywhere that function used to exist is now replaced by `getReachability()` by [this](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8835/files) PR.

Reply:
As discussed offline, the NetworkWatcherImpl changes are not required because QNetworkInformation is already cross-platform.